### PR TITLE
Fix dolt_schemas for doltgres

### PR DESCRIPTION
**Pull request by tbantle22**

### CODE REVIEW

File: go/libraries/doltcore/sqle/database.go [hunk 1]
```diff
@@ -799,9 +799,8 @@ func (db Database) checkForPgCatalogTable(ctx *sql.Context, tableName string) (s
 // case-insensitive manner. The table is returned along with its case-sensitive matched name. An error is returned if
 // no such table exists.
 func (db Database) resolveUserTable(ctx *sql.Context, root doltdb.RootValue, tableName string) (doltdb.TableName, *doltdb.Table, bool, error) {
-	// TODO: dolt_schemas needs work to be compatible with multiple schemas
 	hasDoltPrefix := doltdb.HasDoltPrefix(tableName)
-	if resolve.UseSearchPath && db.schemaName == "" && !hasDoltPrefix {
+	if resolve.UseSearchPath && db.schemaName == "" && (tableName == doltdb.SchemasTableName || !hasDoltPrefix) {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/database.go:803]: Now that you've resolved the question about where the `dolt_schemas` table lives, I think the right move here is to just remove the prefix test entirely, any user space table should resolve the same way.

**Review by zachmu — APPROVED** [review]
LGTM

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>28af51e</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>28af51e</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>ac51a28</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>ac51a28</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**